lib/src/components/sprite_animation_component.dart

```diff
@@ -14,18 +14,24 @@ class SpriteAnimationComponent extends PositionComponent {
   bool removeOnFinish = false;
 
   /// Creates a component with an empty animation which can be set later
-  SpriteAnimationComponent();
+  SpriteAnimationComponent({
+    Vector2 position,
+    Vector2 size,
+    this.animation,
+    this.overridePaint,
+    this.removeOnFinish = false,
+  }) : super(position: position, size: size);
 
   /// Creates an [SpriteAnimationComponent] from an [animation] and a [size]
   ///
   /// Optionally [removeOnFinish] can be set to true to have this component be auto removed from the [BaseGame] when the animation is finished.
+  @Deprecated('Use SpriteAnimationComponent instead')
   SpriteAnimationComponent.fromSpriteAnimation(
     Vector2 size,
     this.animation, {
     this.removeOnFinish = false,
-  }) : assert(animation != null) {
-    super.size.setFrom(size);
-  }
+  })  : assert(animation != null),
```

Comment:
Since size and position were added as optional to the default constructor, I think this factory should work the same way, so `size` should be optional, and a `position` should be added

Reply:
consistency!

Reply:
This is not resolved but this method can be removed like we mentioned before, but don't forget to change the next constructor on this file as well, `fromFrameData` needs to be changed to have both size and position on its optional parameter list.